portal/component-tests/_macros/pagination.component-test.js

```diff
@@ -0,0 +1,288 @@
+    });
+
+    it('should render an ellipsis after the \'First\' and \'Previous\' links', () => {
+      wrapper.expectElement('[data-cy="firstPreviousEllipsis"]').hasClass('govuk-body');
+      wrapper.expectElement('[data-cy="firstPreviousEllipsis"]').hasClass('govuk-!-margin-bottom-3');
+      wrapper.expectText('[data-cy="firstPreviousEllipsis"]').toRead('...');
+    });
+  };
+
+  const shouldNOTRenderFirstAndPreviousLinksAndEllipsis = () => {
+    it('should NOT render a \'First\' link', () => {
+      wrapper.expectElement('[data-cy="First_listItem"]').notToExist();
+
+      wrapper.expectLink('[data-cy="First"]').notToExist();
+    });
+
+    it('should NOT render a \'Previous\' link', () => {
+      wrapper.expectElement('[data-cy="Previous_listItem"]').notToExist();
+
+      wrapper.expectLink('[data-cy="Previous"]').notToExist();
+    });
+
+    it('should NOT render an ellipsis after the \'First\' and \'Previous\' links', () => {
+      wrapper.expectElement('[data-cy="firstPreviousEllipsis"]').notToExist();
+    });
+  };
+
+  const shouldRenderEllipsisAndNextAndLastLinks = (activePage) => {
+    it('should render an ellipsis before the \'Next\' and \'Last\' links', () => {
+      wrapper.expectText('[data-cy="nextLastEllipsis"]').toRead('...');
+      wrapper.expectElement('[data-cy="nextLastEllipsis"]').hasClass('govuk-body');
+      wrapper.expectElement('[data-cy="nextLastEllipsis"]').hasClass('govuk-!-margin-bottom-3');
+    });
+
+    it('should render a \'Next\' link', () => {
+      wrapper.expectElement('[data-cy="Next_listItem"]').hasClass('govuk-body');
+      wrapper.expectElement('[data-cy="Next_listItem"]').hasClass('govuk-!-margin-bottom-3');
+
+      wrapper.expectLink('[data-cy="Next"]').toLinkTo(`/testRoute/${activePage + 1}?testQuery=test`, 'Next');
+      wrapper.expectElement('[data-cy="Next"]').hasClass('govuk-link');
+    });
+
+    it('should render a \'Last\' link', () => {
+      wrapper.expectElement('[data-cy="Last_listItem"]').hasClass('govuk-body');
+      wrapper.expectElement('[data-cy="Last_listItem"]').hasClass('govuk-!-margin-bottom-3');
+
+      wrapper.expectLink('[data-cy="Last"]').toLinkTo('/testRoute/99?testQuery=test', 'Last');
+      wrapper.expectElement('[data-cy="Last"]').hasClass('govuk-link');
+    });
+  };
+
+  const shouldNOTRenderEllipsisAndNextAndLastLinks = () => {
+    it('should NOT render an ellipsis before the \'Next\' and \'Last\' links', () => {
+      wrapper.expectText('[data-cy="nextLastEllipsis"]').notToExist();
+    });
+
+    it('should NOT render a \'Next\' link', () => {
+      wrapper.expectElement('[data-cy="Next_listItem"]').notToExist();
+
+      wrapper.expectLink('[data-cy="Next"]').notToExist();
+    });
+
+    it('should NOT render a \'Last\' link', () => {
+      wrapper.expectElement('[data-cy="Last_listItem"]').notToExist();
+
+      wrapper.expectLink('[data-cy="Last"]').notToExist();
+    });
+  };
```

Comment:
Nice DRY tests - similar to Alex's comment above, should we these into their own directory e.g `test-assertions`?

Also is it worth moving all the `data-cy` selectors into their own variables, similar to cypress selectors?

Reply:
Personally, I think it makes sense for these functions to remain in this test file - they are specific to this pagination component and won't be used anywhere else.

Just to clarify, RE your suggestion about the `data-cy` selectors, do you mean replacing e.g., `[data-cy="Page_${i}_listItem"]` with a function that returns this, e.g., listItemForPage(i)?

Reply:
I've moved the functions into their own directory `test-assertions`.

As discussed on Slack, I've not replaced the `data-cy` selectors with constants, although this could be part of a future refactor